Prolangle/Languages/TypeScript.cs

```diff
@@ -0,0 +1,31 @@
+using Prolangle.Languages.Framework;
+
+namespace Prolangle.Languages;
+
+public class TypeScript : BaseLanguage
+{
+	private TypeScript()
+	{
+	}
+
+	public static TypeScript Instance { get; } = new();
+
+	public override Guid Id { get; } = Guid.NewGuid();
+	public override string Name => "TypeScript";
+	public override TypeSystem Typing => TypeSystem.Duck | TypeSystem.Structural | TypeSystem.Strong | TypeSystem.Static;
+	public override bool Compiled => true;
+	public override MemoryManagement MemoryManagement => MemoryManagement.TracingGarbageCollection;
+	public override SyntaxStyle SyntaxStyle => SyntaxStyle.C;
+
+	public override Applications KnownForBuilding =>
+		Applications.Client | Applications.Configuration |
```

Comment:
I question `Configuration` (within the TS ecosystem, _I guess_, but outside of that?) and `Scripts` (do OSes ship with it? Does it even work without compiling to a different file?)

Reply:
Deno can run TS files directly. But it's not widely used yet...